idaes/models_extra/power_generation/unit_models/soc_submodels/conductive_slab.py

```diff
@@ -0,0 +1,315 @@
+        def dTdz(b, t, ix, iz):
+            return common._interpolate_2D(
+                ic=iz,
+                ifaces=izfaces,
+                nodes=b.znodes,
+                faces=b.zfaces,
+                phi_func=lambda izf: b.temperature[t, ix, izf] / b.length_z[None],
+                phi_bound_0=0,
+                phi_bound_1=0,
+                derivative=True,
+            )
+
+        @self.Expression(tset, ixfaces, iznodes)
+        def heat_flux_x(b, t, ix, iz):
+            return -b.thermal_conductivity * b.dTdx[t, ix, iz]
+
+        @self.Constraint(tset, iznodes)
+        def heat_flux_x0_eqn(b, t, iz):
+            return b.heat_flux_x0[t, iz] == b.heat_flux_x[t, ixfaces.first(), iz]
+
+        @self.Constraint(tset, iznodes)
+        def heat_flux_x1_eqn(b, t, iz):
+            return b.heat_flux_x1[t, iz] == b.heat_flux_x[t, ixfaces.last(), iz]
+
+        @self.Expression(tset, ixnodes, izfaces)
+        def heat_flux_z(b, t, ix, iz):
+            return -b.thermal_conductivity * b.dTdz[t, ix, iz]
+
+        @self.Expression(tset, ixnodes, iznodes)
+        def resistivity(b, t, ix, iz):
+            return (
+                pyo.units.ohm
+                * pyo.units.m
+                * pyo.exp(
+                    b.resistivity_log_preexponential_factor
+                    + b.resistivity_thermal_exponent_dividend / b.temperature[t, ix, iz]
+                )
+            )
+
+        @self.Expression(tset, ixnodes, iznodes)
+        def resistance(b, t, ix, iz):
+            return b.resistivity[t, ix, iz] * b.length_x * b.dx[ix] / b.xface_area[iz]
+
+        @self.Expression(tset, ixnodes, iznodes)
+        def voltage_drop(b, t, ix, iz):
+            return b.current[t, iz] * b.resistance[t, ix, iz]
+
+        @self.Expression(tset, iznodes)
+        def resistance_total(b, t, iz):
+            return sum(b.resistance[t, ix, iz] for ix in ixnodes)
+
+        @self.Expression(tset, iznodes)
+        def voltage_drop_total(b, t, iz):
+            return sum(b.voltage_drop[t, ix, iz] for ix in ixnodes)
+
+        @self.Expression(tset, ixnodes, iznodes)
+        def joule_heating(b, t, ix, iz):
+            # current_density is the current density so have to multiply it be Area I**2 = i**2*A**2
+            # R = rho * dx / Area / (1-porosity) heating = I**2*R
+            return b.current[t, iz] * b.voltage_drop[t, ix, iz]
+
+        @self.Constraint(tset, ixnodes, iznodes)
+        def energy_balance_solid_eqn(b, t, ix, iz):
+            return (
+                b.node_volume[ix, iz] * b.dcedt_solid[t, ix, iz]
+                == b.xface_area[iz]
+                * (b.heat_flux_x[t, ix, iz] - b.heat_flux_x[t, ix + 1, iz])
+                + b.zface_area[ix]
+                * (b.heat_flux_z[t, ix, iz] - b.heat_flux_z[t, ix, iz + 1])
+                + b.joule_heating[t, ix, iz]
+            )
+
+        if dynamic:
+            self.energy_balance_solid_eqn[tset.first(), :, :].deactivate()
+
```

Comment:
Does this model need to overload the `initialize_build` method? At the moment, I think calling `initialize` on this model would result in attribute errors due to the absence of `self.control_volume`. If this model is not intended to be solved/initialized in isolation, I would suggest having `initialize_build` raise a `NotImplementedError("This model is not intended to be initialized in isolation")`